examples/ticks_and_spines/tick_labels_from_values.py

```diff
@@ -0,0 +1,37 @@
+"""
+
+Basic demo showing how to set tick labels to values of a series.
+
+Using ax.set_xticks causes the tick labels to be set on the currently
+chosen ticks. However, you may want to allow matplotlib to dynamically
+choose the number of ticks and their spacing.
+
+In this case may be better to determine the tick label from the value
+at the tick. The following example shows how to do this.
+
+NB: The MaxNLocator is used here to ensure that the tick
+values take integer values. As such, we need to catch
+any IndexErrors in the format function where we have not
+defined a label for that particular tick
+
+"""
+
+import matplotlib.pyplot as plt
+from matplotlib.ticker import FuncFormatter, MaxNLocator
+fig = plt.figure()
+ax = fig.add_subplot(111)
+xs = range(26)
+ys = range(26)
+labels = list('abcdefghijklmnopqrstuvwxyz')
+
+
+def format_fn(tick_val, tick_pos):
+    try:
+        return labels[int(tick_val)]
```

Comment:
@cammil, you are absolutely right that many of matplotlib's demos fall into this trap, and there is ongoing work to address some of that. However, in this situation, I think the negative indices should be considered a bug, not a feature. The code will throw out any ticks greater than 'z', so panning one way gets one effect, while panning the other way gets another.

Maybe a compromise might be to use a modulus operator?